addons/account/models/account_move.py

```diff
@@ -2587,6 +2594,7 @@ def action_switch_invoice_into_refund_credit_note(self):
                         'debit' : line_vals['credit'],
                         'credit' : line_vals['debit']
                     })
+            move.invoice_partner_bank_id # just used so the field is in cache when recomputed before it is written
```

Comment:
I'm not sure what triggers the recompute
`invoice_partner_bank_id` is triggered by `bank_partner_id`
And `bank_partner_id` is triggered by `commercial_partner_id`, 
Then `commercial_partner_id` is triggered by `partner_id`, which doesn't depend on `invoice_line_ids` or `invoice_partner_bank_id`

If the issue is that the field is protected because it is recomputed, wouldn't the solution be to not set the field to `False` in the first place?

Reply:
the field is protected because we are writing on it:

https://github.com/odoo/odoo/blob/550f9e77625a7917bffd693f18aafe10a04276c9/odoo/models.py#L3559-L3570

but before we do the field write on `invoice_partner_bank_id`, we do the field write on 'line_ids' that is causing an account.move.line().unlink() that is causing a flush of everything and a recompute of everything.

Reply:
> I'm afraid that removing breaks that fix.

It wouldn't break the fix if the compute function is called again...

> the field is protected because we are writing on it:

Yes, that I know

> that is causing a flush of everything and a recompute of everything.

AFAIK, stored fields should not be recomputed unless triggered by a field change. I don't see what would trigger it.

________

My understanding with your hack is that indeed you cache the old value, but that value is the wrong one, and we would get back to the behavior before 564c4f5ab13ae45dcb06ccad0de732845a05da61

Reply:
@william-andre , I applied your suggestion, I tested and you were right, removing `False` doesn't break the old fix, thank you :pray: